src/components/structures/UserMenu.tsx

```diff
@@ -292,23 +294,41 @@ export default class UserMenu extends React.Component<IProps, IState> {
                     })}
                 </div>
             )
-        } else if (signupLink) {
+        } else if (signupLink || elementProConfig) {
+            let elementProIFrame;
+            if (elementProConfig && elementProConfig.url) {
+                // If element_pro.domains is set to a non-empty array, only show
+                // dialog if the user is on the domain or a subdomain.
+                const elementProDomains = elementProConfig.domains || [];
+                const mxDomain = MatrixClientPeg.get().getDomain();
+                const validDomains = elementProDomains.filter(d => (d === mxDomain || mxDomain.endsWith(`.${d}`)));
```

Comment:
would be good to mention that we're matching subdomains intentionally here

Reply:
It's mentioned a few lines above but I'll make it clearer.

Reply:
oh, ha - that's probably fine and I'm just blind.